### PR TITLE
Add sonarcloud config

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY

Add sonarcloud config

##### ISSUE TYPE

- Test Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: .sonarcloud.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sonar.sources=lib/ansible/,test/lib/
+sonar.tests=test/
+sonar.test.exclusions=test/lib/**/*,test/support/**/*
+sonar.cpd.exclusions=**/*
+sonar.issue.ignore.multicriteria=python
+sonar.issue.ignore.multicriteria.python.ruleKey=python:*
+sonar.issue.ignore.multicriteria.python.resourceKey=**/*.py
```
Comment (sivel) [inline .sonarcloud.properties:7]: There isn't a ton of documentation available here, but from my understanding from UI tutorials, the `ruleKey` it listed as "Rule Key Pattern", and either matches the name of the rule, or the code, and while I've seen no wildcards used with the code, there are examples with the name. I suppose until this is merged, I have no way of really knowing whether this works unless we go through some test setup.

Comment (sivel) [reply .sonarcloud.properties:7]: https://docs.sonarsource.com/sonarcloud/advanced-setup/analysis-scope/#ignoring-specific-rules-from-specific-files

**Comment by mattclay** [comment]
The ignore settings seem to have no effect. Closing this for now. We can re-open if there's a resolution to that.

File: .sonarcloud.properties (hunk 1 above)
Comment (nitzmahone) [reply .sonarcloud.properties:7]: We spent several hours trying a metric crapton of permutations on those things on a test repo, including reading the Sonarqube sources to understand their custom `.properties` parsing and semantics. We're sure it was "working", in that basics like `sonar.sources` and `sonar.tests` were working, but we were never able to get any form of `ignore` to work properly (individual rules/paths, multiple mapping rulesets with the `multicriteria` key list, wildcards, whatever).

**Pull request closed without merging**